src/hpc/thread.c

```diff
@@ -191,7 +213,19 @@ static NOINLINE void GrowStack(void)
 
 static NOINLINE void SetupTLS(void)
 {
-#ifndef HAVE_NATIVE_TLS
+#if defined USE_MACOS_PTHREAD_TLS_ASM
+    static char asm_code[] = {
+        // movq %gs:(,%rdi,8), %rax
+        // retq
+        0x65, 0x48, 0x8b, 0x04, 0xfd, 0x00, 0x00, 0x00, 0x00, 0xc3
+    };
+    if (memcmp((void *)pthread_getspecific, asm_code, sizeof(asm_code)) != 0) {
+        fprintf(stderr, "internal error: %s\n",
+            "inlining pthread_getspecific() assembly code failed");
+        SyExit(1);
```

Comment:
Please consider using `Panic` here

Reply:
See above, I'm going to remove it entirely.